### DISCUSSION TITLE
Any thoughts on Contentlayer?

### DISCUSSION THREAD

**Discussion by Music47ell** [timlrx/tailwind-nextjs-starter-blog#261] [Ideas]
Any thoughts on [Contentlayer](https://github.com/contentlayerdev/contentlayer)? It can make importing MDX and CMS content a lot easier and it will also live reload content on change. 
It was implemented recently on Lee's blog here https://github.com/leerob/leerob.io/pull/384

**Comment by timlrx** [comment]
Looks pretty interesting. It uses `mdx-bundler` to load mdx contents under the hood as well so it abstracts that part away. Still seems pretty new and in development though - the docs does not even look 25% complete haha. Would prefer to stick with the current solution until the API and docs firm up on Contentlayer side.

In the latest release (v1.1.4) I also introduced a reload on content change. It works slightly differently compared to Contentlayer. It seems like content layer converts the files to js objects which are imported and watched. My current solution establishes a websocket connection which triggers a page reload upon update.

**Reply by vahidtakro** [reply]
The positive thing with Contentlayer is that deployment takes shorter time. and this is amazing for those with a lot of posts. for me now it takes 14 minutes to deploy.

**Reply by Music47ell** [reply]
Yeah, the docs and code needs a lot of work. Like this util file for example: https://github.com/schickling-test/leerob.io/blob/0ccb48eadfcc73ec51e08853707c55df9d5acbd2/lib/utils.ts#L1

I did see the latest changes you made but didn't try them yet.

**Reply by timlrx** [reply]
You tried it out? I would have thought it would take longer since there is an intermediate step where it converts to a js object. Internally it uses the same mdx-bundler. I guess there might be a caching element to speed things up and prevent old files from being compiled to js again - that might help.

The canary release of nextjs should help in the compile time as well. I believe it uses swc instead of webpack, but I will wait for the stable release to be out before doing the upgrades.

**Reply by timlrx** [reply]
> Yeah, the docs and code needs a lot of work. Like this util file for example: https://github.com/schickling-test/leerob.io/blob/0ccb48eadfcc73ec51e08853707c55df9d5acbd2/lib/utils.ts#L1
> 
> I did see the latest changes you made but didn't try them yet.

Yeah I saw that and thought we can wait a while for it to be incorporated. Otherwise, it's hard to maintain the template if the internals keep changing. But, definitely an interesting project to watch.
